mods/ts/rules/defaults.yaml

```diff
@@ -566,6 +569,9 @@
 	Carryable:
 		RequiresCondition: !inside-tunnel
 	RevealOnFire:
+	RejectsOrders@tunnel:
+		Reject: Stop
```

Comment:
Actually, I take that back.  The following should be enough, but will of course need some proper testing:
```diff
diff --git a/OpenRA.Mods.Common/Activities/Move/Move.cs b/OpenRA.Mods.Common/Activities/Move/Move.cs
index e7418d6e8..2ce6e5cda 100644
--- a/OpenRA.Mods.Common/Activities/Move/Move.cs
+++ b/OpenRA.Mods.Common/Activities/Move/Move.cs
@@ -142,7 +142,7 @@ List<CPos> EvalPath()
 
                public override Activity Tick(Actor self)
                {
-                       if (IsCanceled)
+                       if (IsCanceled && mobile.CanInteractWithGroundLayer(self))
                                return NextActivity;
 
                        if (mobile.IsTraitDisabled)
@@ -277,12 +277,6 @@ void SanityCheckPath(Mobile mobile)
                        return Pair.New(nextCell, subCell);
                }
 
-               public override bool Cancel(Actor self)
-               {
-                       path = NoPath;
-                       return base.Cancel(self);
-               }
-
                public override IEnumerable<Target> GetTargets(Actor self)
                {
                        if (path != null)
```